### PR TITLE
Retry flaky test

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

Several tests sometimes fail due to randomness and slower test environment on Github Action.
This change executes retry once if first trial fails. Then failure twice in a row is considered as actual failure.
(e.g. take over 10sec to start processing received packets which causes IDLE.TIMEOUT then shutdown its connection.)

## Testing

See automation....

## Documentation

N/A

### CODE REVIEW

File: scripts/run-gtest.ps1 [hunk 1]
```diff
@@ -295,6 +296,11 @@ function Add-XmlResults($TestCase) {
         $Node = $XmlResults.testsuites.testsuite | Where-Object { $_.Name -eq $TestSuiteName }
     }
     if ($null -ne $Node) {
+        $TestCaseNode = $Node.testcase | Where-Object { $_.Name -eq $TestCaseName }
+        if ($TestCaseNode) {
+            $IsRetry = $true
+            $Node.retried = ($Node.retried -as [Int]) + 1
```
Comment (ami-GS) [inline scripts/run-gtest.ps1:302]: powershell failed at here.
$Node doesn't have `retried` property.
How can I add?

**Comment by ami-GS** [comment]
Error annotation is shown, but not treated as failure by retry

image

File: scripts/run-gtest.ps1 [hunk 2]
```diff
@@ -913,11 +924,12 @@ try {
     Log "$($TestCount) test(s) run."
     if ($KeepOutputOnSuccess -or ($TestsFailed -ne 0) -or ($global:CrashedProcessCount -ne 0)) {
         Log "Output can be found in $($LogDir)"
-        if ($ErrorsAsWarnings) {
+        if ($ErrorsAsWarnings -or
+            (($IsolationMode -eq "Isolated") -and ($TestsFailed -ne 0) -and ($TestsFailed -eq $TestsRetried))) {
             Write-Warning "$($TestsFailed) test(s) failed."
-            Write-Warning "$($TestsFailed) test(s) failed, $($global:CrashedProcessCount) test(s) crashed."
+            Write-Warning "$($TestsFailed) test(s) failed, $($TestsRetried) test(s) retried, $($global:CrashedProcessCount) test(s) crashed."
```
Comment (nibanks) [inline scripts/run-gtest.ps1:930]: What do you think about simplifying this message?
```suggestion
            Write-Warning "$($TestsFailed) test(s) passed with retry."
```

File: scripts/run-gtest.ps1 [hunk 3]
```diff
@@ -913,11 +924,12 @@ try {
...
         } else {
-            Write-Error "$($TestsFailed) test(s) failed, $($global:CrashedProcessCount) test(s) crashed."
+            Write-Error "$($TestsFailed) test(s) failed, $($TestsRetried) test(s) retried, $($global:CrashedProcessCount) test(s) crashed."
```
Comment (nibanks) [inline scripts/run-gtest.ps1:932]: Can we identify how many tests passed on retry and how many failed on retry? We want to highlight the consistent failures.

**Review by nibanks — APPROVED** [review]

**Pull request merged**